DS and Algorithm/DFS.py

```diff
@@ -21,17 +20,19 @@ def peek(self):
 
     def size(self):
         return len(self.items)
-    
+
 def dfs(graph, start):
     visited = set()
     stack = Stack()
-    
     stack.push(start)
+    
     while not stack.is_empty():
         vertex = stack.pop()
+        
         if vertex not in visited:
             visited.add(vertex)
             print(vertex, end=" ")  # Print visited vertex
+            
             # Add all neighbors to the stack
             for neighbor in graph[vertex]:
                 if neighbor not in visited:
```

Comment:
**Issue**: The DFS implementation does not handle cyclic graphs, which can lead to infinite recursion. <br> **Fix**: Add a check to ensure that a vertex is added to the stack only if it has not been visited. <br> **Code Suggestion**: 
 ```
 +                if neighbor not in visited:
 +                    stack.push(neighbor)
 ```